modules/150-user-authn/images/dex/patches/client-groups.patch

```diff
@@ -1,64 +1,21 @@
-diff --git a/server/handlers.go b/server/handlers.go
-index eb65f490..660a01c4 100644
---- a/server/handlers.go
-+++ b/server/handlers.go
-@@ -19,6 +19,7 @@ import (
- 	jose "gopkg.in/square/go-jose.v2"
- 
- 	"github.com/dexidp/dex/connector"
-+	"github.com/dexidp/dex/pkg/groups"
- 	"github.com/dexidp/dex/server/internal"
- 	"github.com/dexidp/dex/storage"
- )
-@@ -491,18 +492,28 @@ func (s *Server) handleApproval(w http.ResponseWriter, r *http.Request) {
- 		return
- 	}
- 
-+	client, err := s.storage.GetClient(authReq.ClientID)
-+	if err != nil {
-+		s.logger.Errorf("Failed to get client %q: %v", authReq.ClientID, err)
-+		s.renderError(r, w, http.StatusInternalServerError, "Failed to retrieve client.")
-+		return
-+	}
-+
-+	if len(client.AllowedGroups) > 0 {
-+		authReq.Claims.Groups = groups.Filter(authReq.Claims.Groups, client.AllowedGroups)
-+		if len(authReq.Claims.Groups) == 0 {
-+			s.logger.Errorf("user not in allowed groups: %v", client.AllowedGroups)
-+			s.renderError(r, w, http.StatusInternalServerError, "User not in allowed groups.")
-+			return
-+		}
-+	}
-+
- 	switch r.Method {
- 	case http.MethodGet:
- 		if s.skipApproval {
- 			s.sendCodeResponse(w, r, authReq)
- 			return
- 		}
--		client, err := s.storage.GetClient(authReq.ClientID)
--		if err != nil {
--			s.logger.Errorf("Failed to get client %q: %v", authReq.ClientID, err)
--			s.renderError(r, w, http.StatusInternalServerError, "Failed to retrieve client.")
--			return
--		}
- 		if err := s.templates.approval(r, w, authReq.ID, authReq.Claims.Username, client.Name, authReq.Scopes); err != nil {
- 			s.logger.Errorf("Server template error: %v", err)
- 		}
+Index: storage/kubernetes/types.go
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
```

Comment:
Can we remove headers for patches? I think it is safe.

Reply:
Of course I removed